include/swift/Parse/Parser.h

```diff
@@ -416,6 +423,14 @@ class Parser {
     return SyntaxContext->finalizeRoot();
   }
 
+  /// Retrieve the token receiver from the parser once it has finished parsing.
+  std::unique_ptr<ConsumeTokenReceiver> takeTokenReceiver() {
+    assert(Tok.is(tok::eof) && "not done parsing yet");
+    auto *receiver = TokReceiver;
```

Comment:
@rintaro Does it make sense to model `TokReceiver` as a `std::unique_ptr` as well? It would make the ownership transfer here a little cleaner to express.

Reply:
I was thinking about that too, unfortunately we currently swap in a pointer to a `DelayedTokenReceiver` on the stack while backtracking.

Reply:
😢